src/js/with-react.jsx

```diff
@@ -108,10 +107,23 @@ const withReact = (WebComponent, { pure = true, passive = false } = {}) => {
 
     render() {
       // eslint-disable-next-line react/prop-types
-      const { props: { children } } = this;
+      const { props: { children }, handleRef } = this;
+
+      // IMPORTANT:
+      // the Custom Element can only be instantiated as soon as it is registered in CustomElementRegistry
+      // which in turn is deferred after DOMReady
+      // hence it's tagName can only be resolved lazily
+      // ref: https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry
+      if (!WCTagName) {
+        const node = new WebComponent();
```

Comment:
Instanciating Custom Elements after the are defined is suuuuper important.
If we try to do it before they will always throw **Invalid Constructor** errors...